flax/linen/linear.py

```diff
@@ -51,6 +54,38 @@ def _canonicalize_tuple(x: Union[Sequence[int], int]) -> Tuple[int, ...]:
     return (x,)
 
 
+def _canonicalize_dtypes(
+    input_dtype: InexactDType,
+    param_dtype: Optional[InexactDType],
+    computation_dtype: Optional[InexactDType]) -> Tuple[InexactDType,
+                                                        InexactDType]:
+  returned_param_dtype = input_dtype if param_dtype is None else param_dtype
```

Comment:
param_dtype should still be float32 by default (see FLIP)

Reply:
Okay, sorry for missing this!

Reply:
Do you think it makes more sense to put the simply make the `param_dtype` default to `float32` on the modules?  That way these helper functions can remain completely type agnostic, and just defer to the input dtype if `None` is passed for the other types.